apps/web/src/features/settings/index.tsx

```diff
@@ -29,6 +32,53 @@ export default function SettingsInterface(): React.ReactNode {
     "",
   );
 
+  // Loading state for save operation
+  const [isSaving, setIsSaving] = useState(false);
+
+  const { session } = useAuthContext();
+  // Handle saving API keys to Supabase
+  const handleSaveApiKeys = async () => {
+    if (!session?.accessToken || !session?.refreshToken) {
+      toast.error("You must be logged in to save API keys");
+      return;
+    }
+    setIsSaving(true);
+
+    try {
+      const apiKeys = {
+        OPENAI_API_KEY: openaiApiKey,
+        ANTHROPIC_API_KEY: anthropicApiKey,
+        GOOGLE_API_KEY: googleApiKey,
+        TAVILY_API_KEY: tavilyApiKey,
+      };
+
+      const response = await fetch("/api/settings/api-keys", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          "x-access-token": session.accessToken,
+          "x-refresh-token": session.refreshToken,
+        },
+        body: JSON.stringify({ apiKeys }),
+      });
+
+      if (!response.ok) {
+        const errorData = await response.json();
+        throw new Error(errorData.error || "Failed to save API keys");
+      }
+
+      await response.json();
+      toast.success("API keys saved successfully");
+    } catch (error) {
+      console.error("Error saving API keys:", error);
+      toast.error(
+        error instanceof Error ? error.message : "Failed to save API keys",
+      );
```

Comment:
dont think instanceof will work in next.js web

need to do a check like this
```suggestion
      let errMessage = "Failed to save API keys";
      if (typeof error === "object" && error && "message" in error && error.message && typeof error.message === "string") {
        errMessage = error.message;
      }
      toast.error(errMessage, { richColors: true });
```